src/cpp/core/include/core/r_util/RActiveSessionStorage.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * RActiveSessionStorage.hpp
+ *
+ * Copyright (C) 2021 by Rstudio, PBC
+ *
+ * Unless you have received this program directly from Rstudio pursuant
+ * to the terms of a commercial license agreement with Rstudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#ifndef CORE_R_UTIL_R_ACTIVE_SESSION_STORAGE
+#define CORE_R_UTIL_R_ACTIVE_SESSION_STORAGE
+
+#include <shared_core/FilePath.hpp>
+
+namespace rstudio {
+namespace core {
+namespace r_util {
+
+   class IActiveSessionStorage 
+   {
+   public:
+      Error virtual readProperty(const std::string& id, const std::string& name, std::string* pValue) = 0;
+      Error virtual writeProperty(const std::string& id, const std::string& name, const std::string& value) = 0;
+
+   protected:
+      virtual ~IActiveSessionStorage() = default;
+      IActiveSessionStorage() = default;
+   };
+
+   class LegacySessionStorage : public IActiveSessionStorage
```

Comment:
Could we call this FileSessionStorage?  It's a little more descriptive and a little less judgmental :)

Reply:
I settled on FileActiveSessionStorage since the base is ActiveSessionStorage, but the spirit is the same, thanks!